quinn/src/send_stream.rs

```diff
@@ -144,8 +144,13 @@ impl SendStream {
             .poll(cx)
             .map(|x| x.unwrap())
         {
-            Poll::Ready(None) => Poll::Ready(Ok(())),
-            Poll::Ready(Some(e)) => Poll::Ready(Err(e)),
+            Poll::Ready(x) => {
+                self.finishing = None;
+                match x {
```

Comment:
Nit: hoist the `Poll::Ready(..)` call?

Reply:
This made a nice concise combinator solution obvious! :tada: